src/test/java/com/opipo/codechallenge/service/TransactionStatusChannelServiceTest.java

```diff
@@ -0,0 +1,100 @@
+        TransactionStatusResponse response = new TransactionStatusResponse();
+        TransactionStatusResponse actual = transactionStatusChannelService.calculateAmounts(channel, response,
+                transactionEntity);
+        assertEquals(actual.getAmount(), amount);
+        assertEquals(actual.getFee(), fee);
+    }
+
+    @Test
+    @DisplayName("Given ATM channel and transactionEntity and transactionStatusResponse then fill it with amout that is amount-fee")
+    void givenAtmAndTransacionEntityAndTransactionStatusResponseThenSetAmount() {
+        ChannelEnum channel = ChannelEnum.ATM;
+        TransactionEntity transactionEntity = Mockito.mock(TransactionEntity.class);
+        BigDecimal amount = BigDecimal.valueOf(14.2d);
+        BigDecimal fee = BigDecimal.valueOf(2.1d);
+        Mockito.when(transactionEntity.getAmount()).thenReturn(amount);
+        Mockito.when(transactionEntity.getFee()).thenReturn(fee);
+        TransactionStatusResponse response = new TransactionStatusResponse();
+        TransactionStatusResponse actual = transactionStatusChannelService.calculateAmounts(channel, response,
+                transactionEntity);
+        assertEquals(actual.getAmount(), amount.subtract(fee));
+    }
+
+    @Test
+    @DisplayName("Given Client channel and transactionEntity and transactionStatusResponse then fill it with amout that is amount-fee")
+    void givenClientAndTransacionEntityAndTransactionStatusResponseThenSetAmount() {
+        ChannelEnum channel = ChannelEnum.CLIENT;
+        TransactionEntity transactionEntity = Mockito.mock(TransactionEntity.class);
+        BigDecimal amount = BigDecimal.valueOf(14.2d);
+        BigDecimal fee = BigDecimal.valueOf(2.1d);
+        Mockito.when(transactionEntity.getAmount()).thenReturn(amount);
+        Mockito.when(transactionEntity.getFee()).thenReturn(fee);
+        TransactionStatusResponse response = new TransactionStatusResponse();
+        TransactionStatusResponse actual = transactionStatusChannelService.calculateAmounts(channel, response,
+                transactionEntity);
+        assertEquals(actual.getAmount(), amount.subtract(fee));
+    }
+
+    @Test
+    @DisplayName("Given ATM channel and transactionEntity without fee and transactionStatusResponse then fill it with amout that is amount")
+    void givenAtmAndTransacionEntityWithoutFeeAndTransactionStatusResponseThenSetAmount() {
+        ChannelEnum channel = ChannelEnum.ATM;
+        TransactionEntity transactionEntity = Mockito.mock(TransactionEntity.class);
+        BigDecimal amount = BigDecimal.valueOf(14.2d);
+        Mockito.when(transactionEntity.getAmount()).thenReturn(amount);
+        TransactionStatusResponse response = new TransactionStatusResponse();
+        TransactionStatusResponse actual = transactionStatusChannelService.calculateAmounts(channel, response,
+                transactionEntity);
+        assertEquals(actual.getAmount(), amount);
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/albertinizao/codechallenge/pullRequest?prid=4681930)